problem/views.py

```diff
@@ -62,10 +63,8 @@ def detail(request, pid):
     except Problem.DoesNotExist:
         logger.warning('problem %s not found' % (pid))
         raise Http404('problem %s does not exist' % (pid))
-    testcase = Testcase.objects.filter(problem=problem)
-    tag = problem.tags.all()
-    return render(request, 'problem/detail.html',
-                  {'problem': problem, 'tags': tag, 'testcase': testcase})
+    problem = get_testcase(problem)
```

Comment:
I prefer 
```
problem.testcase = get_testcase(problem)
def get_testcase(problem)
    return Testcase.objects.filter(problem=problem)
```
or just
```
problem.testcase = Testcase.objects.filter(problem=problem)
```
It's wired to use variable problem to receive return value from get testcase.